source/baseObject.py

```diff
@@ -123,6 +123,29 @@ def invalidateCaches(cls):
 		for instance in cls.__instances.keys():
 			instance.invalidateCache()
 
+class GestureCollector(AutoPropertyType):
+	"""A metaclass used for collecting and caching gestures on a ScriptableObject"""
+
+	def __new__(meta, name, bases, dct):
+		gestures = {}
+		cls = super(GestureCollector, meta).__new__(meta, name, bases, dct)
+		for i in reversed(cls.__mro__):
+			try:
+				gestures.update(getattr(i, "_%s__gestures" % i.__name__))
+			except AttributeError:
+				pass
+		for name, script in dct.iteritems():
+			if not name.startswith('script_'):
+				continue
+			scriptName = name[7:]
+			if hasattr(script, 'gesture'):
+				gestures[script.gesture] = scriptName
+			if hasattr(script, 'gestures'):
+				for gesture in script.gestures:
+					gestures[gesture] = scriptName

```

Comment:
Because we're already looping through the class dict here, we may as well just store the functions rather than the names, saving pointless lookups later. Instead of calling `self.bindGestures(self._gestures)` later, you can just assign `self._gestureMap` to a copy of `self._gestures`. (We still need `_gestureMap` on the instance because someone can call bindGesture at runtime.) I hope this makes sense... :)